src/extensions/CustomDateUtils.tsx

```diff
@@ -0,0 +1,75 @@
+import { DateIOFormats } from '@date-io/core/IUtils';
+import DayjsUtils from '@date-io/dayjs';
+import dayjs, { Dayjs } from 'dayjs';
+
+interface Opts {
+    locale?: string;
+    instance?: typeof dayjs;
+    formats?: Partial<DateIOFormats>;
+  }
+
+  type Constructor<TDate extends Dayjs> = (
+    ...args: Parameters<typeof dayjs>
+  ) => TDate;
+
+const withLocale = <TDate extends Dayjs>(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    dayjs: any,
+    locale?: string,
+): Constructor<TDate> => (!locale ? dayjs : (...args) => dayjs(...args).locale(locale));
+
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const JsonPlugin = (option: any, dayjsClass: any) => {
+    // overriding existing API
+    dayjsClass.prototype.toJSON = function() {
+        return this.format();
```

Comment:
Why does this work? Is it because `dayjs` objects normally use the built-in `Date.prototype.toJSON()` (which outputs a UTC ISO string) and `format()` by default without any arguments outputs some other, timezone-naive format?

Reply:
yes